agent/token/store.go

```diff
@@ -216,6 +224,27 @@ func (t *Store) UpdateReplicationToken(token string, source TokenSource) bool {
 	return changed
 }
 
+// UpdateConfigFileRegistrationToken replaces the current config file registration token
+// in the store. Returns true if it was changed.
+func (t *Store) UpdateConfigFileRegistrationToken(token string, source TokenSource) bool {
+	return t.updateToken(token, source,
+		&t.configFileRegistrationToken, &t.configFileRegistrationTokenSource,
```

Comment:
Since this is a method off `(t *Store)` anyways, can we omit `dstToken *string, dstSource *TokenSource` from the signature and use the `t.configFileRegistrationToken` and `t.configFileRegistrationTokenSource` fields directly in `updateToken`?

Reply:
Ah, I meant to re-refactor the other `Update*Token` methods to use the `updateToken` function I added here.

I added that in https://github.com/hashicorp/consul/pull/15828/commits/f814544fc7172a41d384ba9cc2bd40407d4dbfa5